cloud-nio/cloud-nio-impl-drs/src/main/scala/cloud/nio/impl/drs/DrsPathResolver.scala

```diff
@@ -51,51 +51,50 @@ abstract class DrsPathResolver(drsConfig: DrsConfig, httpClientBuilder: HttpClie
+    val httpPost = makeHttpRequestToMartha(drsPath, fields)
     executeMarthaRequest(httpPost)
   }
 
   /** *
     * Resolves the DRS path through Martha url provided in the config.
     * Please note, this method returns an IO that would make a synchronous HTTP request to Martha when run.
     */
-  def resolveDrsThroughMartha(drsPath: String): IO[MarthaResponse] = {
-    rawMarthaResponse(drsPath).use(httpResponseToMarthaResponse(drsPathForDebugging = drsPath))
+  def resolveDrsThroughMartha(drsPath: String, fields: NonEmptyList[MarthaField.Value]): IO[MarthaResponse] = {
+    rawMarthaResponse(drsPath, fields).use(httpResponseToMarthaResponse(drsPathForDebugging = drsPath))
   }
 }
 
+final case class DrsConfig(marthaUrl: String)
 
-final case class DrsConfig(marthaUri: String, marthaRequestJsonTemplate: String)
+object MarthaField extends Enumeration {
+  val GsUri: MarthaField.Value = Value("gsUri")
+  val Size: MarthaField.Value = Value("size")
+  val TimeCreated: MarthaField.Value = Value("timeCreated")
+  val TimeUpdated: MarthaField.Value = Value("timeUpdated")
+  val GoogleServiceAccount: MarthaField.Value = Value("googleServiceAccount")
+  val Hashes: MarthaField.Value = Value("hashes")
+  val FileName: MarthaField.Value = Value("fileName")
+}
 
-final case class Url(url: String)
-final case class ChecksumObject(checksum: String, `type`: String)
-final case class DosDataObject(name: Option[String],
-                               size: Option[Long],
-                               checksums: Option[Array[ChecksumObject]],
-                               updated: Option[String],
-                               urls: Array[Url])
-final case class DosObject(data_object: DosDataObject)
-final case class SADataObject(data: Json)
+final case class MarthaRequest(url: String, fields: NonEmptyList[MarthaField.Value])
 
-final case class MarthaV2Response(dos: DosObject, googleServiceAccount: Option[SADataObject])
+final case class SADataObject(data: Json)
 
 /**
-  * A response from `martha_v3` or converted from `martha_v2`.
+  * A response from `martha_v3`.
   *
   * @param size Size of the object stored at gsUri
+  * @param timeCreated The creation time of the object at gsUri
   * @param timeUpdated The last update time of the object at gsUri
-  * @param bucket The bucket name port of gsUri, for example "bucket"
-  * @param name The object name part of gsUri, for example "12/345"
   * @param gsUri Where the object bytes are stored, possibly using a generated path name such as "gs://bucket/12/345"
   * @param googleServiceAccount The service account to access the gsUri contents
   * @param fileName A possible different file name for the object at gsUri, ex: "gsutil cp gs://bucket/12/345 my.vcf"
   * @param hashes Hashes for the contents stored at gsUri
   */
 final case class MarthaResponse(size: Option[Long],
+                                timeCreated: Option[String],
```

Comment:
Why extract `timeCreated`? AFAIK it isn't used anywhere no?

Reply:
`timeUpdated` isn't actually used by Cromwell either, but is on the this "cloud-nio / [alfre](https://github.com/broadinstitute/alfre)-super-early-fork" [SPI](https://en.wikipedia.org/wiki/Service_provider_interface).

Out of my sense-of-consistency, I could either remove `timeUpdated` from the implementation of the SPI, or add `timeCreated` to the implementation. But it didn't make sense to only have one field but not both.